### PR TITLE
streamingccl: test that replicated backup fails post cutover

### PR DESCRIPTION
**Pull request by msbutler**
Informs #107089

Epic: none

### CODE REVIEW

File: pkg/ccl/streamingccl/streamingest/datadriven_test.go [hunk 1]
```diff
@@ -378,30 +390,43 @@ func (d *datadrivenTestState) cleanup(t *testing.T) {
...
+		rows, err = d.replicationClusters.DestTenantSQL.DB.QueryContext(ctx, query)
 	default:
 		t.Fatalf("unsupported value to run SQL query as: %s", as)
 	}
 
+	if err != nil {
+		return "", err
+	}
 	output, err := sqlutils.RowsToDataDrivenOutput(rows)
 	require.NoError(t, err)
```
Comment (stevendanna) [inline pkg/ccl/streamingccl/streamingest/datadriven_test.go:415]: Should we return this error now that we have an error return?

File: pkg/ccl/streamingccl/streamingest/datadriven_test.go [hunk 2]
```diff
@@ -105,6 +106,9 @@ func TestDataDriven(t *testing.T) {
 		if strings.HasSuffix(path, ".txt") {
 			return
 		}
+		if !strings.HasSuffix(path, "jobs_post_cutover") {
+			return
+		}
```
[the remark this answers is not in the document]
Comment (msbutler) [reply pkg/ccl/streamingccl/streamingest/datadriven_test.go]: Oof. Yes. Removed.

File: pkg/ccl/streamingccl/streamingest/datadriven_test.go (hunk 1 above)
Comment (stevendanna) [reply pkg/ccl/streamingccl/streamingest/datadriven_test.go:412]: As is works for me.

File: pkg/ccl/streamingccl/streamingest/datadriven_test.go [hunk 3]
```diff
@@ -425,6 +447,16 @@ func (d *datadrivenTestState) execAs(t *testing.T, as, query string) {
 	}
 }
 
+func handleRegex(t *testing.T, err error, regexError string) error {
+	matched, matchStringErr := regexp.MatchString(regexError, err.Error())
+	require.NoError(t, matchStringErr)
+	if matched {
+		return nil
+	} else {
+		return errors.Wrapf(err, "error does not match regex error %s", regexError)
```
Comment (stevendanna) [inline pkg/ccl/streamingccl/streamingest/datadriven_test.go:456]: I think there is another gotcha potential here. If the query _doesn't_ return an error and err is nil then `errors.Wrapf(nil,...)` also returns nil.  In this case, I think you'll fail before that with a NPE at line 451.

Comment (msbutler) [reply pkg/ccl/streamingccl/streamingest/datadriven_test.go:459]: oof. good call. apologies for the general brain fog on this pr.

File: pkg/ccl/streamingccl/streamingest/testdata/jobs_post_cutover [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+exec-sql as=destination-tenant
+SET CLUSTER SETTING jobs.debug.pausepoints = '';
+----
+
+exec-sql as=destination-tenant
+RESUME JOB $backupID
+----
+
+
+# An empty next run indicates the schedule is paused.
```
Comment (stevendanna) [inline pkg/ccl/streamingccl/streamingest/testdata/jobs_post_cutover:44]: Comment seems unrelated

**Review by stevendanna — APPROVED** [review]

**Comment by msbutler** [comment]
TFTR!

bors r=stevendanna

**Comment by msbutler** [comment]
blathers backport 23.2
